src/components/ContentNode/DeviceFrame.vue

```diff
@@ -0,0 +1,135 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+<template>
+  <div
+    class="device-frame"
+    :data-device="device"
+    :class="classes"
+    :style="styles"
+  >
+    <div class="device-screen" :class="{ 'with-device': currentDeviceAttrs }">
+      <slot />
+    </div>
+    <div class="device" />
+  </div>
+</template>
+
+<script>
+import DeviceFrames from 'theme/constants/DeviceFrames.js';
+import { getSetting } from 'docc-render/utils/theme-settings';
+
+const round = (v, dec = 4) => +(`${Math.round(`${v}e+${dec}`)}e-${dec}`);
+
+export default {
+  name: 'DeviceFrame',
+  props: {
+    device: {
+      type: String,
+      required: true,
+    },
+  },
+  provide: {
+    insideDeviceFrame: true,
+  },
+  computed: {
+    currentDeviceAttrs: ({ device }) => getSetting(['theme', 'device-frames', device], DeviceFrames[device]),
+    styles: ({
+      toPixel, toUrl, toPct, currentDeviceAttrs: attrs = {},
+    }) => {
+      const {
+        screenTop, screenLeft, screenWidth, frameWidth,
+        lightUrl, darkUrl, screenHeight, frameHeight,
+      } = attrs;
+
+      return {
+        '--screen-top': toPct(screenTop / frameHeight),
```

Comment:
I'm not sure how you would want to handle it yet, but we should likely add some sort of guard for when someone mistakenly uses dimensions that result in dividing by zero here—otherwise we end up with funky NaN percentage strings.